### ISSUE TITLE
Add code measuring CPU frequency

### ISSUE THREAD

**Issue by Bulat-Ziganshin** [rurban/smhasher#125]
I just wrote a little snippet measuring actual frequency of core performing the code: https://encode.su/threads/3389-Code-snippet-to-compute-CPU-frequency

Please consider using it to correctly compute number of CPU cycles spent by hash functions

**Comment by rurban** [comment]
Nice. Just we already have better measurements than gettimeofday

**Comment by erthink** [comment]
As I wrote earlier, seems that the best code for measuring up to clock cycles [inside the t1ha benchmark](https://github.com/erthink/t1ha/blob/master/tests/mera.c).

It supports x86, arm64, ppc64, s390x, e2k, ia64, etc, as well as `perf_event`, `emscripten_get_now()`, `mach_absolute_time()`, `QueryPerformanceCounter()`, `read_wall_time()`, `clock_gettime()`, `gethrtime()` and `gettimeofday()` (i.e. more than google-benchmark).
For instance see [logs](https://travis-ci.org/github/erthink/t1ha/builds/682728316) on Travis-CI.

I was planning to rearrange this code as a separate "mera" library, but I don't have time for this yet.
Therefore, reusing this code is not as convenient as we would like.
However, it is worth mentioning in this context.

**Comment by Bulat-Ziganshin** [comment]
It seems that you both say about measuring time intervals, while the code I provided is about measuring effective CPU frequency - using any abovementioned way to measure the time interval.

My point is that using rdtsc to co8nt CPU cycles is broken for about 10 years, because it reports cycles of fixed base frequency. So, instead I wrote small code for which we know how much CPU cycles it will be executed, and by measuring time spent on it, we can easily compute the frequency. Moreover, the method works for almost any supersclalar CPU.

Using this approach, we can finally correctly report how much CPU cycles spent for each hashing operation.

**Comment by rurban** [comment]
Yes, I know these loop counting tricks from gamers to calculate the frame rate. It's a rather stable way to do it. I'll check if rtdsc with cpuid is better or worse.

**Comment by rurban** [comment]
Our cycle counter code is correct for Intel. In fact one of the only ones which is actually correct, after an Intel paper.
